hangman.rb

```diff
@@ -0,0 +1,263 @@
+    @num_of_slots = @game_word.length
+    @array_of_slots = []
+
+    @num_of_slots.times do 
+      @array_of_slots.push("_ ")
+    end
+
+    # array of arrays, each is a mistake level drawing
+    @drawings = [[
+              "|     _________",
+              "|     |/      |",
+              "|     |      ",
+              "|     |      ",
+              "|     |       ",
+              "|     |      ",
+              "|     |",
+              "| ____|___"
+            ],
+            [
+              "|     _________",
+              "|     |/      |",
+              "|     |      " + "(_)".colorize(:red),
+              "|     |      ",
+              "|     |       ",
+              "|     |      ",
+              "|     |",
+              "| ____|___"
+            ],
+
+                        [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |       " + "|".colorize(:light_yellow),
+              "|     |       ",
+              "|     |      ",
+              "|     |",
+              "| ____|___"
+            ],
+                                    [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |       |",
+              "|     |       " + "|".colorize(:green),
+              "|     |      ",
+              "|     |",
+              "| ____|___"
+            ],
+                                    [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |      "+ "\\".colorize(:cyan) + "|",
+              "|     |       |",
+              "|     |      ",
+              "|     |",
+              "| ____|___"
+            ],
+                                                [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |      \\|" + "/".colorize(:blue),
+              "|     |       |",
+              "|     |      ",
+              "|     |",
+              "| ____|___"
+            ],
+                                                [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |      \\|/ ",
+              "|     |       | ",
+              "|     |      " + "/ ".colorize(:magenta),
+              "|     |",
+              "| ____|___"
+            ],
+                                                [
+              "|     _________",
+              "|     |/      |",
+              "|     |      (_)",
+              "|     |      \\|/",
+              "|     |       |",
+              "|     |      / " + "\\ ".colorize(:light_black),
+              "|     |",
+              "| ____|___"
+]]
+
+    start
+  end
+
+  def pick_word
+    word_list = %w{cat dog egg horse donkey elephant pig monkey iguana cow goat unigoat eggplant mouse dragon phoenix}
```

Comment:
This would be a good place to use a constant now that we've used them a bit more